gemdaq-testing/gemsupervisor/src/common/tbutils/ThresholdScan.cc

```diff
@@ -193,32 +180,40 @@ bool gem::supervisor::tbutils::ThresholdScan::run(toolbox::task::WorkLoop* wl)
     hw_semaphore_.take();
     
     for (auto chip = vfatDevice_.begin(); chip != vfatDevice_.end(); ++chip) {
+
 	  (*chip)->setRunMode(0);
 	  
 	  hw_semaphore_.give();
 	  wl_semaphore_.give();
+
 	  wl_->submit(readSig_);
 	  
+    }//end for
+
 	  wl_semaphore_.take();
 	  hw_semaphore_.take();
-    }
 	  
-	  //flush FIFO
-    for (int i = 0; i < 2; ++i){
-      if (readout_mask >> i) {
-	glibDevice_->flushFIFO(i);
-	while (glibDevice_->hasTrackingData(i))
-	  std::vector<uint32_t> dumping = glibDevice_->getTrackingData(i);
+  //flush fifo
+  for (int i = 0; i < 2; ++i){
```

Comment:
This possibly explains your problem when flushing the FIFO, in the V2 firmware, there are only 2 GTX links (one for each OptoHybrid)
Depends on how you use readout_mask, which I haven't gone back now to look at.